src/lib/autoinstall/script.rb

```diff
@@ -0,0 +1,388 @@
+        tok = Yast::URL.Parse(newloc)
+        @location = "#{tok["scheme"]}://#{File.join(tok["host"], File.dirname(tok["path"]), path)}"
+      else
+        config = Yast::AutoinstConfig
+        @location = "#{config.scheme}://#{File.join(config.host, config.directory, path)}"
+      end
+      log.info "resolved location #{@location.inspect}"
+    end
+  end
+
+  class ScriptFeedback
+    # feedback settings
+    # @return [:message | :warning | :error | :popup | :no]
+    attr_reader :value
+
+    # @param hash [Hash] hash from which read script feedback value
+    # @option hash [Boolean] "feedback" if feedback is enabled or not
+    # @option hash [String] "feedback_value" how to show feedback
+    def initialize(hash)
+      @value = if hash["feedback"]
+        case hash["feedback_type"]
+        when "message" then :message
+        when "warning" then :warning
+        when "error" then :error
+        when "", nil then :popup
+        else
+          raise "Unknown feedback type value #{hash["feedback_type"].inspect}"
+        end
+      else
+        :no
+      end
+    end
+
+    # serializes back feedback to hash
+    # @return [Hash] for values see {#initialize}
+    def to_hash
+      case value
+      when :message then { "feedback" => true, "feedback_type" => "message" }
+      when :warning then { "feedback" => true, "feedback_type" => "warning" }
+      when :error then { "feedback" => true, "feedback_type" => "error" }
+      when :popup then { "feedback" => true, "feedback_type" => "" }
+      when :no then { "feedback" => false, "feedback_type" => "" }
+      else
+        raise "Unknown value '#{value.inspect}"
+      end
+    end
+  end
+
+  # Scripts that are executed by YaST ( so all except init scripts )
+  class ExecutedScript < Script
+    # Sets if feedback after script finish should be shown
+    # @return [ScriptFeedback]
+    attr_reader :feedback
+
+    # Interpreter to use to run script.
+    # @return [String]
+    attr_reader :interpreter
+
+    # Notification message during script run
+    # @return [String] empty string if no notification should be shown
+    attr_reader :notification
+
+    # Params passed to script
+    # @return [Array]
+    attr_reader :params
+
+    # Reruns script even if already ran
+    # @return [Boolean]
+    attr_reader :rerun
+
+    def to_hash
+      res = super
+      res["interpreter"] = interpreter
+      res["notification"] = notification
+      res["param-list"] = params
+      res["rerun"] = rerun
+      res.merge(feedback.to_hash)
+    end
+
+    def initialize(hash)
+      super
+
+      @feedback = ScriptFeedback.new(hash)
+      @interpreter = hash["interpreter"] || "shell"
+      @notification = hash["notification"] || ""
+      @params = hash["param-list"] || []
+      @rerun = !!hash["rerun"]
+    end
+
+    # mapping of interpreter keywords and its debug flag
+    DEBUG_FLAG_MAP = {
+      "shell" => "-x",
+      "perl"  => "-w",
+      "ruby"  => "-w"
```

Comment:
I guess that if you set the "interpreter" to the full path (e.g., `/usr/bin/perl`) it will not work, right?

Reply:
yes, it won't work. You need to specify "interpreter" to `"/usr/bin/perl -w"` which works.